README.md

```diff
@@ -1,53 +1,290 @@
 
-- `-v /<mydatalocation>:/var/www/html`
+Nextcloud:
+- `/var/www/html/` folder where all nextcloud data lives
+```console
+$ docker run -d nextcloud \
+-v nextcloud:/var/www/html
+```
 
-For fine grained data persistence, you can use 3 volumes, as shown below.
+Database:
+- `/var/lib/mysql` MySQL / MariaDB Data
+- `/var/lib/postresql/data` PostegreSQL Data
+```console
+$ docker run -d mariadb \
+-v db:/var/lib/mysql
+```
 
-- `-v /<mydatalocation>/apps:/var/www/html/apps` installed / modified apps
-- `-v /<mydatalocation>/config:/var/www/html/config` local configuration
-- `-v /<mydatalocation>/data:/var/www/html/data` the actual data of your Nextcloud
+If you want to get fine grained access to your individual files, you can mount additional volumes for data, config, your theme and custom apps. 
+The `data`, `config` are stored in respective subfolders inside `/var/www/html/`. The apps are split into core `apps` (wich are shipped with nextcloud and you don't need to take care of) and a `custom_apps` folder. If you use a custom theme it would go into the `themes` subfolder.
 
-## ... via [`docker-compose`](https://github.com/docker/compose)
+Overview of the folders that can be mounted as volumes:
+
+- `/var/www/html` Main folder, needed for updating
+- `/var/www/html/custom_apps` installed / modified apps
+- `/var/www/html/config` local configuration
+- `/var/www/html/data` the actual data of your Nextcloud
+- `/var/www/html/theme/<YOU_CUSTOM_THEME>` theming/branding
+
+If you want to use named volumes for all of these it would look like this
+```console
+$ docker run -d nextcloud \
+-v nextcloud:/var/www/html \
+-v apps:/var/www/html/custom_apps \
+-v config:/var/www/html/config \
+-v data:/var/www/html/data \
+-v theme:/var/www/html/themes/<YOUR_CUSTOM_THEME>
+```
+
+
+# Running this image with docker-compose
+The easiest way to get a fully featured and functional setup is using a `docker-compose` file. There are too many different possibilities to setup your system, so here are only some examples what you have to look for. 
+
+At first make sure you have chosen the right base image (fpm or apache) and added the features you wanted (see below). In every case you want to add a database container and docker volumes to get easy access to your persistent data. When you want to have your server reachable from the internet adding HTTPS-encryption is mandatory! See below for more information.
+
+## Base version - Apache
+This version will use the apache image and add a mariaDB container **without encryption**. The volumes are set to keep your data persistent.
+
+```yaml
+version: '2'
+
+volumes:
+  nextcloud:
+  db:
+
+services:
+  db:
+    image: mariadb
+    restart: always
+    volumes:
+      - db:/var/lib/mysql
+    environment:
+      - MYSQL_ROOT_PW=...
```

Comment:
I'd remove the `...`, then if you don't put anything, docker-compose throw an error.